internal/kafka/internal/services/clusters_test.go

```diff
@@ -2336,7 +2472,480 @@ func Test_ClusterService_GetExternalID(t *testing.T) {
+	availableStrimziVersions, err := json.Marshal([]api.StrimziVersion{
+		{
+			Version: strimziOperatorVersion,
+			Ready:   true,
+			KafkaVersions: []api.KafkaVersion{
+				{
+					Version: "2.7.0",
+				},
+				{
+					Version: "2.8.0",
+				},
+			},
+			KafkaIBPVersions: []api.KafkaIBPVersion{
+				{
+					Version: "2.7",
+				},
+				{
+					Version: "2.8",
+				},
+			},
+		},
+	})
+
+	if err != nil {
+		t.Fatal("failed to convert available strimzi versions to json")
+	}
+
+	false_availableStrimziVersions, err := json.Marshal([]api.StrimziVersion{
+		{
+			Version: strimziOperatorVersion,
+			Ready:   false,
+		},
+	})
+
+	if err != nil {
+		t.Fatal("failed to convert available strimzi versions to json")
+	}
+
+	mockCluster := &api.Cluster{
+		Meta: api.Meta{
+			ID:        testID,
+			CreatedAt: time.Now(),
+		},
+		Region:                   testKafkaRequestRegion,
+		ClusterID:                testClusterID,
+		CloudProvider:            testKafkaRequestProvider,
+		Status:                   api.ClusterReady,
+		AvailableStrimziVersions: availableStrimziVersions,
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "should return true if Strimzi kafka versions are available in cluster",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory:   &clusters.ProviderFactoryMock{},
+			},
+			args: args{
+				cluster:        mockCluster,
+				strimziVersion: "strimzi-cluster-operator.from-cluster",
+				kafkaVersion:   "2.7.0",
+				ibpVersion:     "2.7",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "Returns false if the provided kafka and kafka ibp versions are not available in the cluster",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory:   &clusters.ProviderFactoryMock{},
+			},
+			args: args{
+				cluster:        mockCluster,
+				strimziVersion: "strimzi-version",
+				kafkaVersion:   "fake-kafka-version",
+				ibpVersion:     "fake-ibpversion",
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "Returns false if kafka version is available in the cluster but the kafka ibp version is not available in the cluster",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory:   &clusters.ProviderFactoryMock{},
+			},
+			args: args{
+				cluster:        mockCluster,
+				strimziVersion: "strimzi-version",
+				kafkaVersion:   "kafka-version",
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "Returns false if kafka ibp version is available in the cluster but the kafka version is not available in the cluster",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory:   &clusters.ProviderFactoryMock{},
+			},
+			args: args{
+				cluster:        mockCluster,
+				strimziVersion: "strimzi-version",
+				ibpVersion:     "ibpversion",
```

Comment:
strimzi and ibp version should be valid

```suggestion
				strimziVersion: strimziOperatorVersion,
				ibpVersion:     "2.7",
```